_docs/tasks/security/health-check.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Enabling Istio CA health check
+overview: This task shows how to enable Istio CA health check.
+
+order: 70
+
+layout: docs
+type: markdown
+---
+{% include home.html %}
+
+This task shows how to enable Istio CA health check. Note this is an alpha feature after Istio V0.6.
+
+After Istio V0.6, Istio CA has a health check feature that can be optionally enabled.
+Currently, the health check feature is able to detect the failures of the CA CSR signing service,
+by periodically sending CSRs to the service. More health check features are coming shortly.
+
+The Istio CA contains a _prober client_ module that periodically checks the CA's status (currently only the health
+status of the gRPC server).
+If the Istio CA is healthy, the _prober client_ updates the _modificate time_ of the health status file
```

Comment:
sort of out of scope for this doc PR but why aren't we using grpc health check instead of this complication/file based approach ?

Reply:
This is mostly because the command line offers better security. K8s probe does not support TLS so we have to open plain text port, which makes the health information public, and makes the CA vulnerable to DoS attacks. I think when K8s probe supports TLS, we can use gRPC health check.

Reply:
afaik we are allowing grpc calls from the mesh and vms, without mtls so not sure I get your point with the benefits